docs/02.API-REFERENCE.md

```diff
@@ -6448,6 +6476,66 @@ jerry_create_arraybuffer (jerry_length_t size);
 - [jerry_value_is_arraybuffer](#jerry_value_is_arraybuffer)
 - [jerry_release_value](#jerry_release_value)
 
+# jerry_arraybuffer_set_external_free_callback
+
+**Summary**
+
+Sets the global callback which is called when the arraybuffer is no longer used.
+It is recommended to set this function before the
+first external arraybuffer is created.
+
+*Note*:
+- When the callback is NULL, no function is called when an external arraybuffer is freed.
+
+**Prototype**
+
+```c
+void jerry_arraybuffer_set_external_free_callback (jerry_external_arraybuffer_free_callback_t callback_p);
+```
+
+- `callback_p` - callback which is called when an external arraybuffer is freed.
+
+*New in version [[NEXT_RELEASE]]*.
+
+**Example**
+
+[doctest]: # ()
+
+```c
+#include "jerryscript.h"
+#include <stdio.h>
+#include <string.h>
+static void
+external_arraybuffer_free_callback (uint32_t length, /**< length of buffer */
+                                    void *buffer, /**< buffer to free (if needed) */
+                                    void *user_p) /**< user pointer */
+{
+  printf ("External arraybuffer is freed!\n");
+}
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+  jerry_arraybuffer_set_external_free_callback (external_arraybuffer_free_callback);
+  const uint32_t buffer_size = 15;
+  const uint8_t base_value = 51;
+  JERRY_VLA (uint8_t, buffer_p, buffer_size);
+  memset (buffer_p, base_value, buffer_size);
+  jerry_value_t external_arraybuffer = jerry_create_arraybuffer_external (buffer_size, buffer_p, NULL);
+  /* The external_arraybuffer_free_callback is called. */
+  jerry_release_value (external_arraybuffer);
```

Comment:
This is a dangerous example as we are passing a stack local memory region for the external arraybuffer. For this example we should use "malloc" and "free" (and not VLA)